specification.markdown

```diff
@@ -245,6 +245,18 @@ A `Binary Element` MUST declare a length in octets from zero to `VINTMAX`.
 
 The contents of a `Binary Element` should not be interpreted by the `EBML Reader`.
 
+# Terminating Elements
+
+`Null Octets`, which are octets with all bits set to zero, MAY follow the value of a `String Element` or `UTF-8 Element` to serve as a terminator. An `EBML Writer` MAY terminate a `String Element` or `UTF-8 Element` with `Null Octets` in order to overwrite a stored value with a new value of lesser length while maintaining the same `Element Data Size` (this can prevent the need to rewrite large portions of an `EBML Document`); otherwise the use of `Null Octets` within a `String Element` or `UTF-8 Element` is NOT RECOMMENDED. An `EBML Reader` MUST consider the value of the `String Element` or `UTF-8 Element` to be terminated upon the first read `Null Octet` and ignore any following data within that `Element`. If an `EBML Writer` terminates the value of a `String Element` or `UTF-8 Element` with a `Null Octet` then any subsequent octets of the `Element` MUST also be `Null Octets`. A string value and a copy of that string value terminated by one or more `Null Octets` are semantically equal.
```

Comment:
> If an `EBML Writer` terminates the value of a `String Element` or `UTF-8 Element` with a `Null Octet` then any subsequent octets of the `Element` MUST also be `Null Octets`.

Missing comma before "then". However, I don't think this is the right thing to do. If we make it a `MUST`, then a reader has to consider a file with e.g. `0x00 0x4a` invalid. I known programmers (I'm one of them). We're lazy. I'm sure there are programs that simply overwrite the first byte of a string with `0x00` and leave the rest as-is, causing such a file to become invalid with the proposed verbiage.

I'd rather say we treat such content the same as C would treat such a string: all bytes starting with the `0x00` byte must be ignored, no matter their content.

As a consequence the following table's "Validity" column should be removed.